File: src/bubbleSort/bubbleSort.py
#!/usr/bin/python
# -*- coding: utf-8 -*-

###################################################
#                       					      
# A simple sorting program using Insertion Sort   
# and vectors in Python 					      
# Author: Jesús Iván Gastelum Romero              
# Contributors: Luis Edmundo Espinoza Larios      
#                       					      
###################################################

from time import time

def bubbleSort(v):
	swaps = True
	while swaps:
		swaps = False
		for i in range(len(v)-1):
			if v[i] > v[i+1]:
				swaps = True
				temp = v[i]
				v[i] = v[i+1]
				v[i+1] = temp
	print v

#*-----------------------------*#
#         MAIN PROGRAM          #
#*-----------------------------*#

start_time = time()
VECTOR = [7 , -5 , 0, 2 , 16, -9, 21, -874, 0, 8, 16, -210]
bubbleSort(VECTOR)
end_time = time() - start_time
print '\nExecution Time: ', end_time, '\n'

File: src/bubbleSort/bubbleSort.c
/*
 * An implementation of the bubble sort algorithm
 * Author Luis Edmundo Espinoza Larios
 * Contributors Jesús Iván Gastelum Romero
 */

#include <stdio.h>
#include <stdlib.h>
#include <time.h>
#include <sys/time.h>
#include "util.h"

/*
 * Sorts an array using the bubble sort algorithm
 */
void sort(int[]);

int main()
{
    static int array[SIZE];
    struct timeval start, end;

    srand(time(NULL));

    generateArray(array);

    gettimeofday(&start, NULL);

    sort(array);

    gettimeofday(&end, NULL);

    printArray(array);

    printf ("Execution time: %f us\n", (double) (end.tv_usec - start.tv_usec) / 1000000 + (double) (end.tv_sec - start.tv_sec));

    return 0;
}

void sort(int array[])
{
    int i, j, temp;

    for (i = 0; i < SIZE - 1; i++) {

        for (j = SIZE - 1; j > i; j--) {

            if (array[j] < array[j - 1]) {
                temp = array[j];
                array[j] = array[j - 1];
                array[j - 1] = temp;
            }
        }
    }
}
